src/AppInstallerSharedLib/Public/AppInstallerVersions.h

```diff
@@ -42,15 +42,15 @@ namespace AppInstaller::Utility
 
         Version() = default;
 
-        Version(const std::string& version, std::string_view splitChars = DefaultSplitChars) :
-            Version(std::string(version), splitChars) {}
-        Version(std::string&& version, std::string_view splitChars = DefaultSplitChars);
+        Version(const std::string& version, std::string_view splitChars = DefaultSplitChars, bool trimPrefix = true) :

```

Comment:
I think I would lean towards having a protected member field that controls the trim behavior and create a custom untrimmed version type.  That might make it harder to stumble into breaking the expectations somewhere, especially if the untrimmed type uses protected inheritance to prevent external use as a "standard" `Version` (at the cost of manually exposing all required methods via `using`).

Reply:
```C++
struct Version {
...
protected:
    bool m_shouldRemovePreambles = true;
...
}

// Name for example
struct PreambleEnabledVersion : protected Version
{
PreambleEnabledVersion() { m_shouldRemovePreambles = false; }
}
```

And if my understanding is correct, then:

```C++
Version a{ "1.0" };
PreambleEnabledVersion b { "1.0" };
// Compilation error as b is not a `Version` outside of its own member functions.
if (a == b) ...
```

All of this is probably over-engineering, but I always try to design against shooting myself in the foot rather than only be careful not to.

Reply:
> All of this is probably over-engineering, but I always try to design against shooting myself in the foot rather than only be careful not to.

I appreciate this; I'd rather do it right the first time than have issues with it later

Reply:
I tried this, and when I use the example as written, I get the message that no overload exists which takes two parameters.  This is coming from the constructor for `PreambleEnabledVersion( std::string&& version, std::string_view splitChars)` not being present.  

I tried creating the constructor, but I think my understanding of how it works is flawed, since it doesn't seem to be properly setting the member field

```
        PreambleEnabledVersion() { m_shouldRemovePreambles = false; }
        
        PreambleEnabledVersion(const std::string& version, std::string_view splitChars = DefaultSplitChars) :
            Version(std::string(version), splitChars) {m_shouldRemovePreambles = false;}
        PreambleEnabledVersion(std::string&& version, std::string_view splitChars = DefaultSplitChars) :
            Version(version, splitChars) {m_shouldRemovePreambles = false;}
```

Reply:
The constructor if the base type will be run before anything in the derived type.  The easiest solution is to just use `Assign` in the derived constructor, rather than invoking the base constructor.